pkg/utils/event.go

```diff
@@ -0,0 +1,114 @@
+package utils
+
+import (
+	"time"
+
+	"github.com/litmuschaos/chaos-runner/pkg/log"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientTypes "k8s.io/apimachinery/pkg/types"
+)
+
+var event EventAttributes
+
+//CreateEvents create the events in the desired resource
+func (engineDetails *EngineDetails) CreateEvents(eventAttributes *EventAttributes, clients ClientSets) error {
+
+	events := &apiv1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      eventAttributes.Name,
+			Namespace: engineDetails.EngineNamespace,
+		},
+		Source: apiv1.EventSource{
+			Component: engineDetails.Name + "-runner",
+		},
+		Message:        eventAttributes.Message,
+		Reason:         eventAttributes.Reason,
+		Type:           eventAttributes.Type,
+		Count:          1,
+		FirstTimestamp: metav1.Time{Time: time.Now()},
+		LastTimestamp:  metav1.Time{Time: time.Now()},
+		InvolvedObject: apiv1.ObjectReference{
+			APIVersion: "litmuschaos.io/v1alpha1",
+			Kind:       "ChaosEngine",
+			Name:       engineDetails.Name,
+			Namespace:  engineDetails.EngineNamespace,
+			UID:        clientTypes.UID(engineDetails.UID),
+		},
+	}
+
+	_, err := clients.KubeClient.CoreV1().Events(engineDetails.EngineNamespace).Create(events)
+	return err
+
+}
+
+//GenerateEvents update the events and increase the count by 1, if already present
+// else it will create a new event
+func (engineDetails *EngineDetails) GenerateEvents(eventAttributes *EventAttributes, clients ClientSets) error {
+
+	var err error
+	event, _ := clients.KubeClient.CoreV1().Events(engineDetails.EngineNamespace).Get(eventAttributes.Reason+string(engineDetails.UID), metav1.GetOptions{})
+	if event.Name != eventAttributes.Name {
+
+		err = engineDetails.CreateEvents(eventAttributes, clients)
+
+	} else {
```

Comment:
we dont need a else, we can return the error from CreateEvents or proceed is its not needed.

Reply:
In that case, we have to write two return statements instead of if-else. anyways made the changes